services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -1236,6 +1239,57 @@ async def _create_fmd_for_upload(
             fmd=fmd
         )
 
+    async def create_s3_export(
+        self,
+        user_id: UserID,
+        object_keys: list[StorageFileID],
+        *,
+        progress_cb: AsyncReportCB | None,
+    ) -> StorageFileID:
+        source_object_keys: set[StorageFileID] = set()
+
+        for object_key in object_keys:
+            async for meta_data_files in get_s3_client(self.app).list_objects_paginated(
+                self.simcore_bucket_name, object_key
+            ):
+                for entry in meta_data_files:
+                    source_object_keys.add(entry.object_key)
+
+        _logger.debug("will archive '%s' files", len(source_object_keys))
+
+        try:
+            destination_object_key = get_random_export_name(user_id)
+
+            await self.create_file_upload_links(
+                user_id=user_id,
+                file_id=destination_object_key,
+                link_type=LinkType.S3,
+                file_size_bytes=ByteSize(0),
+                sha256_checksum=None,
+                is_directory=False,
+            )
+
+            await create_and_upload_export(
+                get_s3_client(self.app),
+                self.simcore_bucket_name,
+                source_object_keys=source_object_keys,
+                destination_object_keys=destination_object_key,
+                progress_cb=progress_cb,
+            )
+        except Exception:  # pylint:disable=broad-exception-caught
+            await self.abort_file_upload(
+                user_id=user_id, file_id=destination_object_key
+            )
+            raise
```

Comment:
who is handling this exception?

Reply:
the caller. in this cased it is OK to fail with a loud exception. You only need to abort the upload incase something went wrong.